client/src/models/footnotes/dynamic_footnotes.js

```diff
@@ -31,85 +27,70 @@ const get_dynamic_footnotes = () => {
     )
   );
   
-    
-  const entities_with_late_planned_spending = _.chain(depts_with_late_planned_spending)
-    .map(Dept.lookup)
-    .flatMap( expand_dept_cr_and_programs )
-    .thru( (late_entities) => late_entities.length > 0 ? [Gov, ...late_entities] : [] )
-    .value();
-  const late_planned_spending_footnotes = _.map(
-    [
-      Gov,
-      ...entities_with_late_planned_spending,
-    ],
-    (subject) => ({
-      subject,
-      topic_keys: ['PLANNED_EXP', 'DP_EXP'],
-      text: `<p>${text_maker(`late_planned_spending_warning_${subject.level}`)}</p>${
-        subject.level === 'gov' ? 
-          `<ul>${
-            _.reduce(
-              depts_with_late_planned_spending,
-              (elements, org_id) => `${elements}<li>${Dept.lookup(org_id).fancy_name}</li>`,
-              ''
+
+  const late_result_or_resource_footnotes = _.flatMap(
+    ['results', 'resources'],
+    (result_or_resource) => {
+      const late_org_property = `late_${result_or_resource}_orgs`;
+
+      const docs_with_late_orgs = _.chain(result_docs_in_tabling_order)
+        .clone() // ...reverse mutates, clone first!
+        .reverse()
+        .filter( (doc) => doc[late_org_property].length > 0 )
+        .value();
+
+      const gov_footnotes = _.map(
+        docs_with_late_orgs,
+        ({ [late_org_property]: late_orgs, doc_type, year }) => ({
+          subject: Gov,
+          topic_keys: result_or_resource === 'resources' ?
+            ['PLANNED_EXP', 'DP_EXP'] :
+            [`${_.toUpper(doc_type)}_RESULTS`],
+          text: `<p>${text_maker(`late_${result_or_resource}_warning_gov`, {result_doc_name: text_maker(`${doc_type}_name`, {year})} )}</p>${
+            `<ul>${
+              _.reduce(
+                late_orgs,
+                (elements, org_id) => `${elements}<li>${Dept.lookup(org_id).fancy_name}</li>`,
+                ''
+              )
+            }</ul>`
+          }`,
```

Comment:
Oof, this is hard to read. Suggest adding a couple of line breaks:

```
text: `<p>
    ${text_maker(`late_${result_or_resource}_warning_gov`, {result_doc_name: text_maker(`${doc_type}_name`, {year})} )}
  </p>
  ${ `<ul>
    ${
      _.reduce(
        late_orgs,
        (elements, org_id) => `${elements}<li>${Dept.lookup(org_id).fancy_name}</li>`,
        ''
      )
    }
  </ul>` }`,
```

Reply:
Haha, thought I could slip that by, I agree it's rough. A bit better with syntax highlighting, but I'll take the suggestion. Thanks!